### DISCUSSION TITLE
How semantic release update a tag version in case of pre release?

### DISCUSSION THREAD

**Discussion by LakshmiRavali** [semantic-release/semantic-release#2245] [Release workflow]
Currently I have a `master` branch that has `package.json` with version `5.30.0` but we have previous tags like 5.29.0, 5.27.0, 5.26.0, 5.25.0 ect.., I have another branch `release-branch` (where I am doing the pre-releases) which has `package.json `with version `5.29.0-rc.1` after master merged to `release-branch`, After that when I do the release from `release-branch` it released `5.25.0-rc.1` I didn't get how this happened?  Can anyone help me understand how pre-release branches release their tags?

Thanks in advance.

**Comment by travi** [comment]
it sounds like history has been modified in your `release-branch`. you likely need to try [these steps](https://semantic-release.gitbook.io/semantic-release/support/troubleshooting#release-not-found-release-branch-after-git-push-force)

**Reply by LakshmiRavali** [reply]
I didn't do any force pushes. Is there anyway I can correct this?

**Reply by LakshmiRavali** [reply]
Can anyone explain me How semantic release creates a tag in case of pre-release. I am in so much confusion .

**Reply by LakshmiRavali** [reply]
How to check on top of which tag it is creating another tag?
